client/sender-app/prompt/prompt.js

```diff
@@ -5,17 +5,54 @@
       .module('app.prompt')
       .controller('Prompt', Prompt);
 
-  Prompt.$inject = [];
+  Prompt.$inject = ['playerMessenger','$state'];
 
-  function Prompt() {
+  function Prompt(playerMessenger, $state) {
     /*jshint validthis: true */
     var vm = this;
     vm.prompt = '';
     vm.submitPrompt = submitPrompt;
 
+    // establishing listener
+    submissionListener();
+
     // example to handle prompt submission
     function submitPrompt() {
+      var prompt = {
+        prompt: vm.prompt
+      };
+      playerMessenger.submit(prompt);
       toastr.info('submitted this prompt: ' + vm.prompt);
     }
+
+    function submissionListener(){
```

Comment:
can we change this function name to 'changeStateListener'?  wanna be consistent across all the controllers with the function we call to establish the listeners in each controller.